rbac/migration_tool/sharedSystemRolesReplicatedRoleBindings.py

```diff
@@ -93,150 +82,116 @@ def set_system_roles(cls):
-        elif default_workspace is None:
-            logger.info(f"Cannot create role binding for role; no resource to bind to: {v1_role.id}")
-        else:
+
+            resource_type = attribute_key_to_v2_related_resource_type(attri_filter["key"])
+            if resource_type is None:
+                # Resource type not mapped to v2
+                continue
+            for resource_id in values_from_attribute_filter(attri_filter):
+                # TODO: Need to bind against "ungrouped hosts" for inventory
+                if resource_id is None:
+                    raise ValueError(f"Resource ID is None for {resource_def}")
+                add_element(perm_groupings, V2boundresource(resource_type, resource_id), v2_perm, collection=set)
+        if default:
             add_element(
-                perm_groupings,
-                V2boundresource(("rbac", "workspace"), default_workspace),
-                v2_perm,
+                perm_groupings, V2boundresource(("rbac", "workspace"), default_workspace), v2_perm, collection=set
             )
-    # Project permission sets to roles per set of resources
-    resource_roles = permission_groupings_to_v2_role_and_resource(perm_groupings, v1_role, binding_mapping)
-    # Construct rolebindings for each resource
-    v2_role_bindings: list[V2rolebinding] = []
-    v2_groups = v1groups_to_v2groups(v1_role.groups)
-    for role, resources in resource_roles.items():
-        for resource in resources:
-            if v2_groups:
-                for v2_group in v2_groups:
-                    if binding_mapping:
-                        role_binding_id = binding_mapping.find_role_binding_by_v2_role(role.id)
-                    else:
-                        role_binding_id = str(uuid.uuid4())
-                    v2_role_binding = V2rolebinding(
-                        role_binding_id, v1_role, role, frozenset({resource}), frozenset({v2_group})
-                    )
-                    v2_role_bindings.append(v2_role_binding)
-            else:
-                if binding_mapping:
-                    role_binding_id = binding_mapping.find_role_binding_by_v2_role(role.id)
-                else:
-                    role_binding_id = str(uuid.uuid4())
-                v2_role_binding = V2rolebinding(role_binding_id, v1_role, role, frozenset({resource}), v2_groups)
-                v2_role_bindings.append(v2_role_binding)
-    return frozenset(v2_role_bindings)
-
 
-custom_roles_created = 0
+    # Project permission sets to roles per set of resources
+    return permission_groupings_to_v2_role_bindings(perm_groupings, v1_role, role_bindings)
 
 
-def permission_groupings_to_v2_role_and_resource(
-    perm_groupings: Permissiongroupings, v1_role: V1role, binding_mapping: Optional[BindingMapping]
-) -> dict[V2role, list[V2boundresource]]:
-    """
-    Determine V2 roles and resources they apply to from a set of V1 resources and permissions.
+def permission_groupings_to_v2_role_bindings(
```

Comment:
This is the "main" change of the MR: multiple mappings, one per v2 role-resource are now computed, and we do not reuse a system role.